scripts/code_style_rewrite_branch.py

```diff
@@ -0,0 +1,214 @@
+    # Check for staged files
+    result = subprocess.run(["git", "diff", "--cached", "--exit-code"], \
+            stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        return False
+    return True
+
+def delete_branch(branch_name) -> bool:
+    """
+    Delete the given branch.
+    """
+    # Go to detached HEAD state in case we are deleting the checked-out branch
+    result = subprocess.run(["git", "checkout", "-d"], check=False)
+    if result.returncode != 0:
+        print("Error detaching HEAD.", file=sys.stderr)
+        return False
+    result = subprocess.run(["git", "branch", "-D", branch_name], check=False)
+    if result.returncode != 0:
+        print("Error deleting branch '" + branch_name + "'.", file=sys.stderr)
+        return False
+    return True
+
+def rebase_on(branch: str, commit: str, new_branch: str) -> bool:
+    """
+    Make a new branch with the given name that consists of the given branch
+    rebased on the given commit.
+    """
+    # Create a new branch from the existing branch
+    result = subprocess.run(["git", "checkout", branch], check=False)
+    if result.returncode != 0:
+        print("Error checking out branch '"+branch+"'.", file=sys.stderr)
+        return False
+    result = subprocess.run(["git", "checkout", "-b", new_branch], check=False)
+    if result.returncode != 0:
+        print("Error creating rebase destination branch '"+new_branch+"'.", \
+                file=sys.stderr)
+        return False
+
+    # Rebase the branch atop the given commit
+    result = subprocess.run(["git", "rebase", commit], check=False)
+    if result.returncode != 0:
+        print("Error rebasing branch '"+branch+"' on commit '"+commit+"'.", \
+                file=sys.stderr)
+        delete_branch(new_branch)
+        return False
+
+    return True
+
+def restyle_commit_onto_current(commit_hash: str) -> bool:
+    """
+    Re-style the given commit and add the result as a new commit on the given
+    branch.
+    """
+    # Get a list of changed files in the commit
+    result = subprocess.run(["git", "diff", "--name-only", commit_hash, \
+            commit_hash + "~"], stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        print("Error getting changed files for commit " + commit_hash + \
+                ".", file=sys.stderr)
+        return False
+    changed_files = str(result.stdout, "ascii").strip().split()
+
+    # Checkout changed files to their state in the old commit
+    for changed_file in changed_files:
+        result = subprocess.run(["git", "checkout", commit_hash, "--", \
+                changed_file], check=False)
+        if result.returncode != 0:
+            print("Error checking out changed file '" + changed_file + \
+                    "' to commit " + commit_hash + ".", file=sys.stderr)
+            return False
+
+    # Restyle the changed files to the new style
+    if code_style.fix_style(changed_files) != 0:
```

Comment:
Blocker: `changed_files` includes non-C files (`.py`, `.data`, etc.). Asking uncrustify to restyle them doesn't go well.

Reply:
Ack. I've just discovered the same issue in parallel, I'll fix that shortly.

Reply:
(I believe) fixed with all of the edge cases sorted out.